tests/validation/tests/v3_api/test_hpa.py

```diff
@@ -0,0 +1,533 @@
+    hpa, workload = edit_hpa(cluster_member_client, ns)
+
+    # Verify that cluster member cannot edit the hpa created by cluster owner
+    verify_edit_forbidden(cluster_member_client, remove_resource)
+
+    remove_resource(hpa)
+    remove_resource(workload)
+    remove_resource(user_project)
+
+
+def verify_hpa_project_readonly_edit(remove_resource):
+    project = rbac_get_project()
+    user_token = rbac_get_user_token_by_role(PROJECT_READ_ONLY)
+    readonly_user_client = get_project_client_for_token(project, user_token)
+    # Verify that read -only user cannot edit the hpa created by cluster owner
+    verify_edit_forbidden(readonly_user_client, remove_resource)
+
+
+def verify_edit_forbidden(user_client, remove_resource,
+                          cluster_owner_client=None, ns=None):
+    metrics = [{
+        'name': 'cpu',
+        'type': 'Resource',
+        'target': {
+            'type': 'Utilization',
+            'utilization': '50',
+        },
+    }]
+
+    if(cluster_owner_client is None and ns is None):
+        project = rbac_get_project()
+        ns = rbac_get_namespace()
+        cluster_owner_token = rbac_get_user_token_by_role(CLUSTER_OWNER)
+        cluster_owner_client = \
+            get_project_client_for_token(project, cluster_owner_token)
+
+    # Create HPA as a cluster owner
+    hpa, workload = create_hpa(cluster_owner_client, ns)
+    # Verify editing HPA fails
+    with pytest.raises(ApiError) as e:
+        user_client.update(hpa,
+                           name=hpa['name'],
+                           namespaceId=ns.id,
+                           maxReplicas=10,
+                           minReplicas=3,
+                           workload=workload.id,
+                           metrics=metrics)
+    assert e.value.error.status == 403
+    assert e.value.error.code == 'Forbidden'
+
+    remove_resource(hpa)
+    remove_resource(workload)
+
+
+def create_hpa(p_client, ns, workload=None):
+
+    # Create workload of scale 1 with CPU reservation
+    # Create hpa pointing to the workload.
+    if workload is None:
+        workload = create_workload(p_client, ns)
+    name = random_test_name("hpa")
+
+    metrics = [{'name': 'cpu',
+                'type': 'Resource',
+                'target': {
+                    'type': 'Utilization',
+                    'utilization': '50',
+                },
+                }]
+
+    hpa = p_client.create_horizontalPodAutoscaler(
+        name=name,
+        namespaceId=ns.id,
+        maxReplicas=5,
+        minReplicas=2,
+        workloadId=workload.id,
+        metrics=metrics
+    )
+    hpa = wait_for_hpa_to_active(p_client, hpa)
+    assert hpa.type == "horizontalPodAutoscaler"
+    assert hpa.name == name
+    # After hpa becomes active, the workload scale should be equal to the
+    # minReplicas set in HPA object
+    workloadlist = p_client.list_workload(uuid=workload.uuid).data
+    assert workloadlist[0].scale == hpa.minReplicas
```

Comment:
@soumyalj Can you assert the value of hpa.minReplicas is set to expected value right after it gets created ?

Reply:
Validate that the workload actually has correct number of pods by calling validate_workload

Reply:
Done